### PR TITLE
doc-stacks: add stacks structure

### PR DESCRIPTION
**Pull request by mhamwala**

### CODE REVIEW

File: docs/stacks/stack-structure.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+---
+title: "Stack Structure"
+path: /docs/stacks/stack-stucture
+section: Stacks
+---
+## Stack structure
+```
+my-stack
+├── README.md
+├── stack.yaml
```
Comment (helenmasters) [inline docs/stacks/stack-structure.md]: The changes look good to me, for completeness you should include a description of `stack.yaml` as you've included all the other files.

File: docs/stacks/stack-structure.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+├── image/
+|   ├── project/
+|   |   └── Dockerfile
+│   └── Dockerfile-stack
+└── templates/
+    └── my-starter/
```
Comment (neeraj-laad) [inline docs/stacks/stack-structure.md]: lets not use the term starter at all

File: docs/stacks/stack-structure.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        └── .appsody-config.yaml
+```
+
+## Image
```
Comment (neeraj-laad) [inline docs/stacks/stack-structure.md]: Could we reword this section to say something on these lines and remove the image as a section:

```Appsody application stacks are provided to developers as a docker image and includes a pre-configured technology stack ready to start application development. ```

**Comment by mhamwala** [comment]
changes addressed

File: docs/stacks/stack-structure.md [hunk 4]
```diff
@@ -13,13 +13,24 @@ my-stack
...
+    └── sample-hello/
+            └── .appsody-config.yaml
 ```
 
-## Image
+## Stack.yaml
+The stack.yaml defines different components of the stack and which template the stack should use by default. See the example below:
+```bash
+    id: sample-stack # must be unique, will be used by the CLI
+    name: Sample Application Stack # one liner description short meaningful name for the stack
```
Comment (helenmasters) [inline docs/stacks/stack-structure.md]: I'd split your words so :
id: sample-stack # short meaningful name for the stack, must be unique, will be used by the CLI
name: concise one line description of the stack

File: docs/stacks/stack-structure.md [hunk 5]
```diff
@@ -13,13 +13,24 @@ my-stack
...
+    version: 0.1.0 # version of the stack
+    description: sample stack to help creation of more appsody stacks # free form text explaining more about the capabilities of this stack and various templates
+    maintainer: John Smith <example@example.com> # comma separated list for more than one maintainer
+    default-template: sample-hello # name of default template
+    license: Apache-2.0 # license for the stack
+```
 
-The image provides foundational capabilities that can be extended or replaced by the user application that is based on it. The image has mechanisms to control which aspects can and cannot be overridden by the developer.
+## Stack Image
+Appsody application stacks are provided to developers as a docker image and include a pre-configured technology stack ready to start application development. It also has mechanisms to control which aspects can and cannot be overridden by the developer.
```
Comment (helenmasters) [inline docs/stacks/stack-structure.md]: docker -> Docker

File: docs/stacks/stack-structure.md (hunk 4 above)
Comment (mhamwala) [reply docs/stacks/stack-structure.md]: I agree

**Pull request merged**